### ISSUE TITLE
Syntax error in SELECT Statement

### ISSUE THREAD

**Issue by JoachimEck** [SAP/abap-cleaner#251]
Hello ,
I just wanted to use the ABAP Cleaner in a programme again. Unfortunately I get an error after a change and the ABAP Cleaner terminates. 

![image](https://github.com/SAP/abap-cleaner/assets/101569831/e934f798-faf5-4475-a964-d340befc3633)
At this line the statement is: 
![image](https://github.com/SAP/abap-cleaner/assets/101569831/d3dfdd45-a6ae-46bb-976c-6894367ab404)

`SELECT aufnr, lead_aufnr, objnr, plnbez, stlbez
FROM caufv
INTO ( @DATA(lv_aufnr), @DATA(lv_lead), @DATA(lv_objnr), @<ls_auftrag>-matnr, @DATA(lv_stlbez) )
WHERE aufnr = @<ls_auftrag>-aufnr
    AND werks = @s_werks
    AND loekz = @space.`

Perhaps it is due to the declarations within the SELECT, but the programme can be activated. 
Can you tell me if it is me or the ABAP Cleaner?

Many thanks
Joachim

**Comment by JoachimEck** [comment]
Hey,
if I swap the variables, it works:
![image](https://github.com/SAP/abap-cleaner/assets/101569831/5d858ad8-d822-446b-b82d-c327b6e779c9)

Can you please check why it doesn't work in the other way.

Thanks,
Joachim

**Comment by JoachimEck** [comment]
Hey Jörg-Michael,
thanks for the quick reply.
Here is my profile, which I use:
[profile.zip](https://github.com/SAP/abap-cleaner/files/14328522/profile.zip)

Kind regards,
Joachim

**Comment by jmgrassau** [comment]
Hi Joachim, 

hm, unfortunately, I still can't reproduce the issue, even with the profile. You are using the current release 1.13.1 (menu "Help / About"), right? Does the error log (menu "Help / Open Error Log") contain any more details? 

Maybe this issue depends on the context, too. In that case, unfortunately, I'd have to ask for some more work on your end… Could you please:
* copy the complete code from the ADT editor to any text editor (e.g. Notepad++)
* in ADT, open menu "Source Code / Show Read-Only Preview With ABAP Cleaner …" 
* you can now use Ctrl+C / Ctrl+V to paste code from the text editor to ABAP cleaner multiple times, trying to reduce the code to a minimum example that triggers the error

Kind regards,
Jörg-Michael

**Comment by JoachimEck** [comment]
Hi Jörg-Michael,

I am happy if I can help. I shrunk everything down as far as I could and the error is still there.
![image](https://github.com/SAP/abap-cleaner/assets/101569831/688fadfd-593f-4f83-bf6a-11563c10d020)
`REPORT.
DATA gt_auftrag      TYPE STANDARD TABLE OF resb.
PARAMETERS s_werks TYPE werks_d OBLIGATORY.
START-OF-SELECTION.
MODULE status_0100 OUTPUT.
  LOOP AT gt_auftrag ASSIGNING FIELD-SYMBOL(<ls_auftrag>).
      SELECT aufnr, lead_aufnr, objnr, plnbez, stlbez
        FROM caufv
        INTO ( @DATA(lv_aufnr), @DATA(lv_lead), @DATA(lv_objnr),
               @<ls_auftrag>-matnr, @DATA(lv_stlbez) )
        WHERE aufnr = @<ls_auftrag>-aufnr
          AND werks = @s_werks
          AND loekz = @space.
      ENDSELECT.
ENDLOOP.
ENDMODULE.`

I have tried it outside the MODULE. It works there, so that could be the reason. Is MODULE not supported?

Kind regards,
Joachim

**Comment by jmgrassau** [comment]
Hi Joachim, 

ah, great, now I get the error message, too! Thanks for your help, let's see what causes this issue! 

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Joachim, 

oh, this is too bad! It's the "Align SELECT lists" rule, which tries to keep the maximum line length of 100. So, an even shorter example (no context needed) would be:

```ABAP
    SELECT f1, f2, f3
      FROM any_dtab
      INTO ( @DATA(lv_var_1_with_long_name), @DATA(lv_var_2_with_very_long_name), @DATA(lv_var_3_with_long_name) ).
    ENDSELECT.
```

which is changed to

```ABAP
    SELECT f1, f2, f3
      FROM any_dtab
      INTO ( @DATA(lv_var_1_with_long_name), @DATA(lv_var_2_with_very_long_name), @DATA(
           lv_var_3_with_long_name) ).
    ENDSELECT.
```

Kind regards,
Jörg-Michael

**Comment by JoachimEck** [comment]
Hi Jörg-Michael,

sorry, i dont get it yet. I've changed the MODULE to an IF-clause. The Error is still there. I understand that it is also because he has to indent in so far. In your example, I don't understand why ABAP Cleaner indents the third varible like this. If I separate the variable from the DATA in my code, I can no longer activate the programme and I get an error.

The line length from the profile is counted from the beginning of the line, isn't it?

For me to understand, is the behaviour so desired and would I need to adjust my profile or are you doing some "magic" ;)?

Kind regards
Joachim

**Comment by jmgrassau** [comment]
Hi Joachim, 

sorry for the confusion: My example was just demonstrating the bug: ABAP cleaner tries to limit the line length to 100 in the INTO clause (as configured in the profile), but it introduces a syntax error by breaking the line after ```@DATA(```, rather than keeping the inline declaration together. So that is the bug which I'll have to fix, since of course, ```@DATA(``` must not be detached from the variable name. 

Kind regards,
Jörg-Michael

**Comment by JoachimEck** [comment]
Hi Jörg-Michael,

got it! Thanks!

Kind regards,
Joachim

**Comment by JoachimEck** [comment]
Hi Jörg-Michael,

that was quick! 
Thanks.

Kind regards,
Joachim

**Comment by jmgrassau** [comment]
Hi Joachim, 

thanks again for reporting this and for your help in narrowing down the issue! This bug should now be resolved with version 1.13.2, which was just released.

Kind regards,
Jörg-Michael
